src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,45 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Contact from which Person, Email and Address inherit 
+ */
+public abstract class Contact {
```

Comment:
Why the choice of abstract class? This limits your ability to instantiate a Contact class. Can you justify your choice?